simplified-main/src/main/java/org/nypl/simplified/main/MainFragmentListenerDelegate.kt

```diff
@@ -213,10 +205,6 @@ internal class MainFragmentListenerDelegate(
         this.openErrorPage(event.parameters)
         state
       }
-      is CatalogBookDetailEvent.DownloadWaitingForExternalAuthentication -> {
```

Comment:
I may not have gotten to it yet, but why is this no longer present?

In the original code, these branches were only reachable when you clicked a book that required a SAML login. They stayed in that state until the login had been completed.

Reply:
I moved this logic to the `MainFragment` because even if several fragments refer to the same book, the action should be performed only once.